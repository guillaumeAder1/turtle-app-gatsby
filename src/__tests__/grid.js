import React from 'react';
import renderer from 'react-test-renderer';

import Grid from '../components/game/grid';
const mines = [{ x: 2, y: 3 }, { x: 3, y: 3 }];
const size = { w: 12, h: 10 };
const turtle = { x: 1, y: 1 };
describe('Grid', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = renderer.create(
      <Grid size={size} turtle={turtle} mines={mines} />,
    );
  });
  it('renders correctly', () => {
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('transformPos should exists', () => {
    expect(wrapper).not.toBe(null);

    expect(wrapper.transformPos).not.toBe(null);
  });
  it('should return val', () => {
    // expect(wrapper.instance().transformPos({ x: 1, y: 1 })).not.toBe('1_1');
  });
});
