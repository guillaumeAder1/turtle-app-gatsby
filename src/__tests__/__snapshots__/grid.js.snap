// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders correctly 1`] = `
Array [
  <div
    className="grid"
  >
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        Q
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        X
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        X
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
  </div>,
  <div>
    <button
      onClick={[Function]}
    >
      Move
    </button>
    <button
      onClick={[Function]}
    >
      Rotate
    </button>
  </div>,
]
`;

exports[`Header renders correctly 1`] = `
Array [
  <div
    className="grid"
  >
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        Q
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        X
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        X
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
    <div
      className="row-container"
    >
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
      <div
        className="cell"
        onClick={[Function]}
      >
        
      </div>
    </div>
  </div>,
  <div>
    <button
      onClick={[Function]}
    >
      Move
    </button>
    <button
      onClick={[Function]}
    >
      Rotate
    </button>
  </div>,
]
`;
